### PR TITLE
Fixes in unity framework

### PR DESCRIPTION
**Pull request by nordic-krch**
Fixes for issues reported on devzone:
- add support for `inline static` functions
- add support for mocking syscalls
- add support for mocking headers without functions

### CODE REVIEW

**Comment by nordic-krch** [comment]
Compliance warning is expected and should be accepted as it complains about `inline static`.

File: scripts/unity/header_prepare.py [hunk 1]
```diff
@@ -39,9 +39,9 @@ def header_prepare(in_file, out_file, out_wrap_file):
     content = syscall_pattern.sub(r"", content)
 
     syscall_decl_pattern = re.compile(
-        r'__syscall\s+(?:\w+[*\s]+)+(.+?;)',
+        r'__syscall\s+((?:\w+[*\s]+)+(.+?;))',
         re.M | re.S)
-    content = syscall_decl_pattern.sub("", content)
+    content = syscall_decl_pattern.sub(r"\1", content)
```
Comment (trond-snekvik) [inline scripts/unity/header_prepare.py:44]: Just strip out `__syscall`, and treat these functions as any other instead?

**Review by trond-snekvik — APPROVED** [review]
I think the syscall logic can be simplified, but LGTM.

File: scripts/unity/header_prepare.py (hunk 1 above)
Comment (nordic-krch) [reply scripts/unity/header_prepare.py]: done

**Comment by nordic-krch** [comment]
Added stripping of inline syscalls, e.g. `z_impl_foo`.

**Review by trond-snekvik — APPROVED** [review]

**Review by tejlmand — APPROVED** [review]

**Pull request merged**